### ISSUE TITLE
turbo:load not fired after turbo:click when using anchor to navigate on the same page

### ISSUE THREAD

**Issue by overdrivemachines** [hotwired/turbo#876]
In my rails app, I have a few links on a page that are used to navigate within the page using anchor tags' href="#idname" as shown:
```erb
<%= link_to "Home", root_path(anchor: "home") %>
<%= link_to "About", root_path(anchor: "about") %>
<%= link_to "Portfolio", root_path(anchor: "portfolio") %>
<%= link_to "Login", login_path %>
```
When I click on Login everything works as expected. The turbo:click event fires and then turbo:load fires, just as expected. However, when I click on Home or About when I'm already on the root_path, then things don't work as expected. The turbo:click event fires and nothing else fires. Can we change this behavior? In my app, whenever there is a turbo:click event a loader image appears and it disappears when the turbo:load event fires. With only turbo:click event firing, my app is stuck showing a loader image.

**Comment by marcoroth** [comment]
Hey @overdrivemachines, thanks for opening this issue!

I don't think it 100% makes sense to also emit a `turbo:load` event in that case. The `turbo:load` event is just fired when a Turbo visit is being made, which is not the case here. 

In this case it's just operating on the current page, as no new data is being fetched. I think if anything, then the `turbo:click` event shouldn't even fire in that case, since Turbo isn't involved.

Does that make sense?

**Comment by tannakartikey** [comment]
@overdrivemachines what if you disable `turbo` for these links? Will that solve your issue?


> In this case it's just operating on the current page, as no new data is being fetched. I think if anything, then the turbo:click event shouldn't even fire in that case, since Turbo isn't involved.

@marcoroth this is something to think about.

**Comment by overdrivemachines** [comment]
> @overdrivemachines what if you disable `turbo` for these links? Will that solve your issue?

I could disable turbo for those links but what if I'm in the login url then I want turbo enabled because otherwise there will be a full page reload when the user wants to navigate to the home page.

**Comment by overdrivemachines** [comment]
> In this case it's just operating on the current page, as no new data is being fetched. I think if anything, then the `turbo:click` event shouldn't even fire in that case, since Turbo isn't involved.

I agree. Maybe it is designed like that for a reason. In the mean time, how do I figure out if the user is operating on the current page and no new data is fetched. For eg if I'm on the root_url and I click on root_url#about. Here is some of the code:
```js
document.addEventListener("turbo:click", () => {
  // how to check if the user is operating on the current page and no new data is going to be fetched

});
```

**Comment by marcoroth** [comment]
The most primitive way would be:
```js
document.addEventListener("turbo:click", event => {
  if (event.target.href.startsWith("#")) {
    // current page
  } else {
    // other page
  }
})
```

**Comment by domchristie** [comment]
> how do I figure out if the user is operating on the current page

@overdrivemachines when navigating to same-page anchors, Turbo performs a "silent" visit. This ensures that the back button works as expected, but it doesn't fire all the events for a normal application visit. However the `hashchange` event is fired so you may find this useful.

**Comment by scratchoo** [comment]
while playing with anchor, I noticed a weird thing

I also have a bunch of urls with a hash # like this:

localhost:3000/home#hash1
localhost:3000/home#hash2
localhost:3000/home#hash3

clicking these urls doesn't fire turbo:load (similar to the post above which I think is the expected behavior)

but if I visit the home path with a query string like this:  localhost:3000/home?query=string

then I click the the link with #hash1 the turbo:load is fired
then if I click the other 2 links with #hash2 and #hash3 turbo:load is not fired

BUT then if I click the link with #hash1 again the turbo:load is fired, and it's keep firing whenever I click it, however the other links (#hash2 and #hash3) never fire turbo:load

so to summarize, if you click a link with anchor after having a query string in your url, the turbo:load will be fired and continue to do so whenever you click that link.

is this a bug?

**Comment by domchristie** [comment]
@scratchoo I've not managed to reproduce the case you describe, however I did notice the following issue:

1. load `/#hash1`
2. click on a link to `#hash2`: `turbo:load` not fired
3. click on a link to `#hash1`: `turbo:load` fired
4. click on a link to `#hash1`: `turbo:load` fired

**Comment by scratchoo** [comment]
@domchristie yes this is similar to what I did above, if you load some #hash then whenever you revisit that link the turbo:load will be fired (only for that specific link). So yeah
